src/bin/pgcopydb/table-data.c

```diff
@@ -945,16 +998,33 @@ copydb_copy_data_by_oid(CopyDataSpec *specs, PGSQL *src, PGSQL *dst,
 		/*
 		 * 1. Now COPY the TABLE DATA from the source to the destination.
 		 */
+		CopyProgressContext context = {
+			.specs = specs,
+			.tableSpecs = tableSpecs,
+			.bytesTransmitted = 0,
+			.lastSavingTimeMs = 0
+		};
+		INSTR_TIME_SET_CURRENT(context.lastSavingTimeMs);
```

Comment:
Can we deal with that responsibility from within the hook function, allowing/expecting the context value to be set to zero by the caller?

Reply:
Just to understand your reasoning: This pattern is quite common in the codebase. What is it that sets this piece of code apart form others?

Reply:
Which pattern do you mean? I can't remember of another place where we use INSTR TIME portability macros for high-precision time measurement from libpq and also a callback function and a context...

Please use the INSTR TIME macros only in the callback function, so that the caller does not need to know why your callback implementation decided to use INSTR TIME macros instead of `time(NULL)` for a 5 seconds duration where accuracy is not a concern, allowing us to change our mind later about how the callback function is implemented without having to also change the code from the caller that has to prepare the context. 

A good init value would be zero, because in C `name = { .specs = specs }` will leave the other fields properly initialized to zero. It is very easy in the callback function to initialize the lastSavingTimeMs to _now_ when called the first time.